src/test/java/com/checkmarx/jira/JiraTestUtils.java

```diff
@@ -0,0 +1,337 @@
+            CustomAsynchronousJiraRestClientFactory factory = new CustomAsynchronousJiraRestClientFactory();
+            URI jiraURI = null;
+            try {
+                jiraURI = new URI(jiraProperties.getUrl());
+            } catch (URISyntaxException e) {
+                //log.error("Error constructing URI for JIRA", e);
+            }
+            this.client = factory.createWithBasicHttpAuthenticationCustom(jiraURI, jiraProperties.getUsername(), jiraProperties.getToken(), jiraProperties.getHttpTimeout());
+
+        }
+    }
+
+    @Override
+    public void deleteIssue(String issueKey) {
+        client.getIssueClient().deleteIssue(issueKey, true).claim();
+    }
+
+    private SearchResult search(String jql) {
+        return  client.getSearchClient().searchJql(jql).claim();
+    }
+
+    @Override
+    public void cleanProject(String projectKey) {
+        SearchResult searchResult = search(String.format("project = \"%s\"", projectKey));
+        for (Issue issue: searchResult.getIssues()) {
+            deleteIssue(issue.getKey());
+        }
+    }
+
+    @Override
+    public int getNumberOfIssuesInProject(String projectKey) {
+        SearchResult result = search(String.format("project = \"%s\"", projectKey));
+        return result.getTotal();
+    }
+
+    @Override
+    public Map<Filter.Severity, Integer> getIssuesPerSeverity(String projectKey) {
+        Map<Filter.Severity, Integer> result= new HashMap<>();
+        SearchResult searchResults = search(String.format("project = \"%s\"", projectKey));
+        for (Issue issue: searchResults.getIssues()) {
+            String severity = getIssueSeverity(issue.getDescription()).toUpperCase();
+            Filter.Severity filterSeverity = Filter.Severity.valueOf(severity.toUpperCase());
+            if (severity != null) {
+                if (result.containsKey(filterSeverity)) {
+                    result.put(filterSeverity,result.get(filterSeverity) + 1 );
+                } else {
+                    result.put(filterSeverity, 1);
+                }
+            }
+        }
+        return result;
+    }
+
+    private String getIssueSeverity(String issueDescription) {
+        return getIssueBodyPart(issueDescription,"Severity:");
+    }
+
+    private String getIssueBodyPart(String issueDescription, String field) {
+        String[] lines = issueDescription.split(System.lineSeparator());
+        for (String line: lines) {
+            if (line.contains(field)) {
+                return line.split(" ")[1];
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public String getIssueFilename(String projectKey) {
+        Issue issue = getFirstIssue(projectKey);
+        String firstLine = issue.getDescription().split(System.lineSeparator())[0];
+        String[] firstLineParts= firstLine.split(" ");
+        return firstLineParts[4];
+    }
+
+    // TODO UDI: check if this is realy the vulnerability field !
```

Comment:
Is the TODO resolved?

Reply:
Well, not yet.
I think this is the most important comment.
I will check it now.

Reply:
Ok. it is resolved.
I made sure it really is the vu,erability.